lambda/es-proxy-layer/lib/query.js

```diff
@@ -176,7 +183,6 @@ async function get_hit(req, res) {
         minimum_confidence_score: _.get(req,'_settings.ALT_SEARCH_KENDRA_FAQ_CONFIDENCE_SCORE'),
         qnaClientFilter: _.get(req, 'session.QNAClientFilter')
     };
-    var no_hits_question = _.get(req, '_settings.ES_NO_HITS_QUESTION', 'no_hits');
     var response = await run_query(req, query_params);
     qnabot.log("Query response: ", JSON.stringify(response,null,2));
     var hit = _.get(response, "hits.hits[0]._source");
```

Comment:
On line `112` in `lambda/es-proxy-layer/lib/handler.js`,  you have `es_response.hits.hits = []`. Here the `._source` is accessed in the first element .  

if `es_response.hits.hits = []`, the `var hit = _.get(response, "hits.hits[0]._source");` will throw the exception 
```
TypeError: Cannot read property '_source' of undefined
```
Will the above line be executed if `es_response.hits.hits = []`?

Reply:
Thanks Taruk. Using lodash defends against non-existent keys and array indexes, so even if `response.hits.hits = []` the expression `_.get(response, "hits.hits[0]._source")` will just evaluate to `undefined`.